### DISCUSSION TITLE
Code Editor and Commit

### DISCUSSION THREAD

**Discussion by vwkd** [community/community#3210] [Mobile] [answered]
I'd love to be able to write simple code right through the app and be able to create commits.

For example, I have a repo for all my notes. Unlike other note taking apps, Git allows me to keep my edit history of my notes transparently. A code editor with commit function would allow me to write my notes directly to my repo, instead of needing to transfer them manually from my note app to VSCode on my desktop.

It's essentially a simple text editor with a git client built-in. The commit function can be really simple, like just asking for a commit message and committing the current changes. More abilities like staging (git add) would be cool but not critical for the beginning.

I know there is work being done on Codespaces, which brings VSCode to the browser. However this is a full-fledged IDE with a backend to run the code (therefore paid) and also not optimized for mobile.

Would love to see you integrate a simple code editor and git client right in the Github app.

[PolyGit](http://www.polygitapp.com/) and [Working Copy](https://workingcopyapp.com/) are iOS apps that can do this currently. Would love to see the GitHub app evolve into that direction.

Edit: Seems I'm not the first. #3161 #3164 #3033 #3155

**Comment by rpxs** [comment]
[Gitpod](https://gitpod.io) has something like this and it integrates with GitHub nicely. Its free to an extent, then you have to pay. Codespaces will bring something similar to this, so until then, Gitpod seems like what you want. This is a really cool idea, and if Codespaces is going to be a paid code editor, then a simple editor where you can commit and add a message sounds good :)

**Comment by spaltrowitz** [comment, marked as answer]
thanks for your suggestion @vwkd! this has now shipped https://github.blog/changelog/2022-11-15-github-for-ios-edit-files-from-browse-code
